src/Orleans.Core/Core/GrainCallFilterExtensions.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Hosting
+{
+    public static class GrainCallFilterExtensions
+    {
+        /// <summary>
+        /// Adds an <see cref="IIncomingGrainCallFilter"/> to the filter pipeline.
+        /// </summary>
+        /// <param name="services">The service collection.</param>
+        /// <param name="filter">The filter.</param>
+        /// <returns>The service collection.</returns>
+        [Obsolete("Use ISiloHostBuilder." + nameof(AddIncomingGrainCallFilter), error: true)]
+        public static IServiceCollection AddGrainCallFilter(this IServiceCollection services, IIncomingGrainCallFilter filter)
+        {
+            throw new NotSupportedException($"{nameof(AddGrainCallFilter)} is no longer supported. Use ISiloHostBuilder.AddIncomingGrainCallFilter(...) instead.");
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IIncomingGrainCallFilter"/> to the filter pipeline.
+        /// </summary>
+        /// <typeparam name="TImplementation">The filter implementation type.</typeparam>
+        /// <param name="services">The service collection.</param>
+        /// <returns>The service collection.</returns>
+        [Obsolete("Use ISiloHostBuilder." + nameof(AddIncomingGrainCallFilter), error: true)]
+        public static IServiceCollection AddGrainCallFilter<TImplementation>(this IServiceCollection services)
+            where TImplementation : class, IIncomingGrainCallFilter
+        {
+            throw new NotSupportedException($"{nameof(AddGrainCallFilter)} is no longer supported. Use ISiloHostBuilder.AddIncomingGrainCallFilter(...) instead.");
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IIncomingGrainCallFilter"/> to the filter pipeline via a delegate.
+        /// </summary>
+        /// <param name="services">The service collection.</param>
+        /// <param name="filter">The filter.</param>
+        /// <returns>The service collection.</returns>
+        [Obsolete("Use ISiloHostBuilder." + nameof(AddIncomingGrainCallFilter), error: true)]
+        public static IServiceCollection AddGrainCallFilter(this IServiceCollection services, GrainCallFilterDelegate filter)
+        {
+            throw new NotSupportedException($"{nameof(AddGrainCallFilter)} is no longer supported. Use ISiloHostBuilder.AddIncomingGrainCallFilter(...) instead.");
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IIncomingGrainCallFilter"/> to the filter pipeline.
+        /// </summary>
+        /// <param name="services">The service collection.</param>
+        /// <param name="filter">The filter.</param>
+        /// <returns>The service collection.</returns>
+        internal static IServiceCollection AddIncomingGrainCallFilter(this IServiceCollection services, IIncomingGrainCallFilter filter)
```

Comment:
please can  you change these to `public` as they are currently not usable (same for the below)

Reply:
@stephenlautier, I'll consider it since this seems to be a popular request. Can you use the methods on ISiloHostBuilder instead?

Reply:
Ahh oki, wasn't aware there was another way of doing it. My bad misread the Obsolete message and tried to use the internal instead. So its good enough, thanks!